iib/workers/tasks/opm_operations.py

```diff
@@ -295,196 +295,132 @@ def inner(*args, **kwargs):
-    :raises AddressAlreadyInUse: if the specified port is already being used by another service.
+    Run OPM render, which can be communicated with using gRPC queries.
+
+    :param str base_dir: temp directory where opm will be executed.
+    :param str input_data: input data for opm render
+        Example: catalog-image | catalog-directory | bundle-image | bundle-directory | sqlite-file
+    :return: list of bundle images parsed from input data
+    :rtype: list(dict)
     """
-    from iib.workers.tasks.utils import run_cmd, terminate_process
-
-    log.debug('Run command %s with up to %d retries', ' '.join(serve_cmd), max_tries)
-    for _ in range(max_tries):
-        rpc_proc = subprocess.Popen(
-            serve_cmd,
-            cwd=cwd,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            universal_newlines=True,
-        )
-        start_time = time.time()
-        while time.time() - start_time < wait_time:
-            time.sleep(5)
-            ret = rpc_proc.poll()
-            # process has terminated
-            if ret is not None:
-                if not rpc_proc.stderr:
-                    raise IIBError(
-                        f'Command "{" ".join(serve_cmd)}" has failed, stderr was not captured'
-                    )
-                stderr_message = rpc_proc.stderr.read()
-                if 'address already in use' in stderr_message:
-                    raise AddressAlreadyInUse(f'Port {port} is already used by a different service')
-                raise IIBError(
-                    f'Command "{" ".join(serve_cmd)}" has failed with error "{stderr_message}"'
-                )
+    from iib.workers.tasks.utils import run_cmd
 
-            # query the service to see if it has started
-            try:
-                output = run_cmd(
-                    ['grpcurl', '-plaintext', f'localhost:{port}', 'list', 'api.Registry']
-                )
-            except IIBError:
-                output = ''
+    log.info("Get list of bundles from %s", input_data)
 
-            if 'api.Registry.ListBundles' in output or 'api.Registry.ListPackages' in output:
-                log.debug('Started the command "%s"; pid: %d', ' '.join(serve_cmd), rpc_proc.pid)
-                log.info('Index registry service has been initialized.')
-                return rpc_proc
+    input_data_path = _get_input_data_path(input_data, base_dir)
 
-        terminate_process(rpc_proc)
+    cmd = [Opm.opm_version, 'render', input_data_path]
 
-    raise IIBError(f'Index registry has not been initialized after {max_tries} tries')
+    opm_render_output = run_cmd(
+        cmd, {'cwd': base_dir}, exc_msg=f'Failed to list bundles on {input_data}'
+    )
+
+    if not opm_render_output:
+        log.info("There are no bundles in %s", input_data)
+        return []
+
+    log.debug("Parsing data from opm render")
+    opm_data = [json.loads(package) for package in re.split(r'(?<=})\n(?={)', opm_render_output)]
```

Comment:
I believe this piece of code is duplicated from `get_list_packages`. Wouldn't make sense to have a private or lower level function just to do this and share the code between `get_list_packages` and `get_list_bundles`?

Reply:
Good idea, changed the code - please check.